packages/flame/lib/src/experimental/layout_component.dart

```diff
@@ -0,0 +1,147 @@
+import 'package:flame/components.dart';
+import 'package:flame/experimental.dart';
+import 'package:flame/src/game/notifying_vector2.dart';
+import 'package:flutter/widgets.dart';
+
+enum Direction { horizontal, vertical }
+
+/// Super class for [RowComponent] and [ColumnComponent]
+/// A relayout is performed when
+///  - a change in this component's children takes place
+///  - the size of this component or a parent changes
+///  - the [gap] parameter is changed
+abstract class LayoutComponent extends PositionComponent with HasGameRef {
+  LayoutComponent(
+    this.direction,
+    this.mainAxisAlignment,
+    this._gap,
+    Vector2? size,
+  ) : super(
+          size: size,
+        ) {
+    isManuallySized = size != null;
```

Comment:
This doesn't have to be done in the body, it can be done directly after the arguments and then the variable doesn't have to be late.